src/ReactiveUI/ObservableAsPropertyHelper.cs

```diff
@@ -26,7 +26,7 @@ public sealed class ObservableAsPropertyHelper<T> : IHandleObservableErrors, IDi
         T _lastValue;
         readonly IConnectableObservable<T> _source;
         IDisposable _inner;
-        private int _activated;
+        private long _activated;
```

Comment:
Is this necessary? I get why you've changed the type to `long` because `Interlocked.Read` requires it, but the reason there is no `int` overload is because a 32-bit read is guarateed to be atomic, no? And besides, does it really matter if `IsSubscribed` isn't 💯 bang up to date? Even with an `Interlocked` call, you'll still have a race condition if you do any meaningful work in response to `IsSubscribed` and multiple threads are in play...

At worst I would think `_activated` could just be marked as `volatile`...

Reply:
Yeah I was debating that during the submission. Left the volatile out for the moment unless you can think of a good reason to add? As you said the C# standard does say int's are atomic read operations.

Reply:
I think _technically_ the use of `volatile` is required on some architectures to ensure that `_activated` is read from main memory rather than from a cache, and therefore being possibly stale. However, in practice I don't know what architectures these are and am not qualified to know the scenarios in which this would actually bite.

However, I do think the `IsSubscribed` property is merely indicative and not something people should be encouraged to use anyway, so I don't see `volatile` as necessary (but again, I could be easily swayed by an argument from those who know better).